respect-lib-shared/src/commonMain/kotlin/world/respect/shared/viewmodel/report/list/ReportListViewModel.kt

```diff
@@ -0,0 +1,254 @@
+import world.respect.shared.domain.report.model.ReportOptions
+import world.respect.shared.domain.report.model.ReportPeriodOption
+import world.respect.shared.domain.report.model.ReportSeries
+import world.respect.shared.domain.report.model.ReportSeriesVisualType
+import world.respect.shared.domain.report.model.ReportSeriesYAxis
+import world.respect.shared.domain.report.model.ReportXAxis
+import world.respect.shared.domain.report.model.StatementReportRow
+import world.respect.shared.domain.report.query.RunReportUseCase
+import world.respect.shared.navigation.NavCommand
+import world.respect.shared.navigation.ReportDetail
+import world.respect.shared.navigation.ReportEdit
+import world.respect.shared.viewmodel.RespectViewModel
+import world.respect.shared.viewmodel.app.appstate.FabUiState
+import world.respect.shared.viewmodel.app.appstate.LoadingUiState.Companion.NOT_LOADING
+import kotlin.time.Clock
+import kotlin.time.ExperimentalTime
+
+data class ReportListUiState(
+    val reportList: () -> PagingSource<Int, Report> = { EmptyPagingSource() },
+    val addSheetOrDialogVisible: Boolean = false,
+    val activeUserPersonUid: Long = 0L,
+    val xAxisFormatter: GraphFormatter<String>? = null,
+    val yAxisFormatter: GraphFormatter<Double>? = null
+)
+
+class ReportListViewModel(
+    savedStateHandle: SavedStateHandle,
+) : RespectViewModel(savedStateHandle) {
+
+    // TODO: Implement proper dependency injection for these use cases
+    // Currently unimplemented - will need to be initialized before use
+    private lateinit var runReportUseCase: RunReportUseCase
+    private val createGraphFormatterUseCase: CreateGraphFormatterUseCase = CreateGraphFormatterUseCase()
+
+    private val _uiState = MutableStateFlow(ReportListUiState())
+    val uiState: Flow<ReportListUiState> = _uiState.asStateFlow()
+    private val activeUserPersonUid: Long = 0
+    private val pagingSourceFactory: () -> PagingSource<Int, Report> = { DummyReportPagingSource() }
+
+    init {
+        _appUiState.update { prev ->
+            prev.copy(
+                navigationVisible = true,
+                title = "reports",
+                fabState = FabUiState(
+                    text = "reports",
+                    icon = FabUiState.FabIcon.ADD,
+                    onClick = this::onClickAdd,
+                )
+            )
+        }
+        _appUiState.update { prev ->
+            prev.copy(
+                fabState = prev.fabState.copy(
+                    visible = true
+                )
+            )
+        }
+        viewModelScope.launch {
+            _uiState.update { prev ->
+                prev.copy(
+                    reportList = pagingSourceFactory,
+                    activeUserPersonUid = activeUserPersonUid
+                )
+            }
+        }
+    }
+
+    @OptIn(ExperimentalTime::class)
+    fun runReport(report: Report): Flow<RunReportUseCase.RunReportResult> = flow {
```

Comment:
Create a new data class for the FlowType - that should include the RunReportResult, xAxisFormatter, and yAxisFormatter.

then:
```
return runReportUseCase(options).map {
    //Construct the data class here
}
```